aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -116,6 +116,31 @@ TORCH_META_FUNC(argmin)
   check_argmax_argmin(*this, "argmin", self, dim, keepdim);
 }
 
+TORCH_META_FUNC(cumsum)
+(const Tensor& self, int64_t dim, c10::optional<ScalarType> dtype) {
+  // Checking whether 'dim' is valid.
+  maybe_wrap_dim(dim, self.dim());
+
+  const auto& result = maybe_get_output();
+  ScalarType out_dtype;
+
+  if (result.defined()) {
+    out_dtype = dtype.value_or(result.scalar_type());
+    // This check is still here because the inline version of structured kernels
+    // does not do any checks on 'set_output'.
```

Comment:
I'm not sure I understand. We do test that the passed dtype matches the out dtype; sample generated code:

```
    void set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides,
                    TensorOptions options, DimnameList names) override {
        
        const auto& out = outputs_[output_idx].get();
        TORCH_CHECK(options.dtype() == out.dtype(),
            "Expected out tensor to have dtype ", options.dtype(), ", but got ", out.dtype(), " instead");        TORCH_CHECK(options.device() == out.device(),
            "Expected out tensor to have device ", options.device(), ", but got ", out.device(), " instead");     
        bool resized = at::native::resize_output(outputs_[output_idx], sizes);

```

Reply:
Ah. Replace `inline` => `inplace`. In the inplace version, we don't test it, do we?

Reply:
Oh yeah, inplace still not testing; see also https://github.com/pytorch/pytorch/issues/57827